src/DataObjectAnnotator.php

```diff
@@ -70,31 +75,44 @@ public function __construct()
      */
     protected function setEnabledClasses($supportedParentClass)
     {
-        foreach((array)ClassInfo::subclassesFor($supportedParentClass) as $class)
-        {
+        foreach ((array)ClassInfo::subclassesFor($supportedParentClass) as $class) {
             $classInfo = new AnnotateClassInfo($class);
-            if($this->permissionChecker->moduleIsAllowed($classInfo->getModuleName())) {
+            if ($this->permissionChecker->moduleIsAllowed($classInfo->getModuleName())) {
                 $this->annotatableClasses[$class] = $classInfo->getClassFilePath();
             }
         }
     }
 
     /**
-     * @param $moduleName
-     * @return array
+     * @return boolean
      */
-    public function getClassesForModule($moduleName)
+    public static function isEnabled()
     {
-        $classes = array();
+        return (bool)static::config()->get('enabled');
+    }
 
-        foreach ($this->annotatableClasses as $class => $filePath) {
-            $classInfo = new AnnotateClassInfo($class);
-            if($moduleName === $classInfo->getModuleName()) {
-                $classes[$class] = $filePath;
-            }
-        }
+    /**
+     * @param boolean $enabled
+     */
+    public static function setEnabled($enabled)
+    {
+        static::config()->enabled = $enabled;
```

Comment:
I'm not sure we need this. If the use case is enabling or disabling the module via \_config.php then it can be done with the configuration directly. SilverStripe is trying to move away from static methods and using the global state to store stuff in favour of class and service instances via injector

Reply:
Oh, I see - it's moved up from lower down in the class. If this will be released as a new major version perhaps we can remove these setters now.